pixeltable/catalog/catalog.py

```diff
@@ -422,6 +422,125 @@ def create_view(
         self._tbls[view._id] = view
         return view
 
+    @_retry_loop
+    def create_replica(self, path: Path, md: list[schema.FullTableMd], if_exists: IfExistsParam) -> Table:
+        """
+        Creates table, table_version, and table_schema_version records for a replica with the given metadata.
+        The metadata should be presented in standard "ancestor order", with the table being replicated at
+        list position 0 and the (root) base table at list position -1.
+        """
+        # If this table UUID already exists in the catalog, it's an error
+        # TODO: Handle the case where it already exists as an anonymous table (because it was an ancestor of
+        #     a different replica)
+        tbl_id = UUID(md[0].tbl_md.tbl_id)
+        if tbl_id in self._tbls:
+            raise excs.Error(
+                f'That table has already been replicated as {self._tbls[tbl_id]._path()!r}. \n'
+                f'Drop the existing replica if you wish to re-create it.'
+            )
+
+        existing = self._handle_path_collision(path, View, False, if_exists)
+        if existing is not None:
+            assert isinstance(existing, View)
```

Comment:
In the collision case, it probably makes sense to insist it's the identical replica/uuid

Reply:
Done.